client/src/components/WorkflowInvocationState/WorkflowInvocationState.vue

```diff
@@ -107,14 +107,17 @@ const invocationStateSuccess = computed(() => {
 
 const workflowStore = useWorkflowStore();
 
-const isDeletedWorkflow = computed(() => getWorkflow()?.deleted === true);
-const workflowVersion = computed(() => getWorkflow()?.version);
+const workflow = ref<Workflow | undefined>(undefined);
+const isDeletedWorkflow = computed(() => workflow.value?.deleted === true);
+const workflowVersion = computed(() => workflow.value?.version);
 
 onMounted(async () => {
     try {
         await invocationStore.fetchInvocationForId({ id: props.invocationId });
-        initialLoading.value = false;
+        invocationLoaded.value = true;
         if (invocation.value) {
+            await workflowStore.fetchWorkflowForInstanceId(invocation.value.workflow_id);
```

Comment:
If all function calls in an async function are awaited, is it even async ? 

Also in this case `useWorkflowInstance` would be much preferable. If you want to improve the error handling there you could easily return a third error in https://github.com/galaxyproject/galaxy/blob/dev/client/src/composables/useWorkflowInstance.ts#L16

Reply:
> If all function calls in an async function are awaited, is it even async ?

Yes it is. The rest of the event loop continues to execute while the execution of this specific function is waiting.

In fact, the opposite is true:

If no function call in an async function is awaited, it is not really async.

Reply:
> Yes it is. The rest of the event loop continues to execute while the execution of this specific function is waiting.

Correct me if I'm wrong, but the calls here will be executed linearly, just like in a sync function (yes, event loop continues) ? None of these calls however need to wait on each other, they're completely independent.

Reply:
> Correct me if I'm wrong, but the calls here will be executed linearly, just like in a sync function

Which is the purpose of async functions. If no await is used, there is no need for the calling function to be async.

> None of these calls however need to wait on each other

the next line depends on this line.

Reply:
> the next line depends on this line

... is what I understood by using await calls as well, hence, I used them here as well.

Reply:
This is the only line, and it can be replaced by `useWorkflowInstance `

Reply:
So we're all on the same page that, if `useWorkflowInstance` is used, there's no need for this to be an async function, because nothing needs to be awaited.

Reply:
The reason I didn't just place `useWorkflowInstance` outside of the `onMounted` hook and the if statement is because when this component is initialized, we have no invocation, and hence, no `invocation.workflow_id`, and so `useWorkflowInstance(invocation.value?.workflow_id)` causes an error (no workflow id).
Therefore, I added an `await fetchInvocationForId` and once that gives us the invocation, we fetch the workflow

I could replace that individual line:
```suggestion
            const { workflow: workflowIns } = useWorkflowInstance(props.invocation.workflow_id);
            workflow.value = workflowIns.value;
```

Reply:
I could also potentially set `workflow` as a `computed` ref like so:
```
const workflow = computed(() => {
    if (invocation.value) {
        const { workflow } = useWorkflowInstance(invocation.value.workflow_id);
        return workflow.value;
    }
    return undefined;
});
const isDeletedWorkflow = computed(() => workflow.value?.deleted === true);
const workflowVersion = computed(() => workflow.value?.version);

onMounted(async () => {
    try {
        await invocationStore.fetchInvocationForId({ id: props.invocationId });
        invocationLoaded.value = true;
        if (invocation.value) {
            await pollStepStatesUntilTerminal();
            await pollJobStatesUntilTerminal();
        }
    } catch (e) {
        errorMessage.value = errorMessageAsString(e);
    } finally {
        initialLoading.value = false;
    }
});
```

but the reason I had it the way it was in the singular mounted hook for fetching invocation AND workflow, was because we were catching errors from either there. That's why I changed `invocation` from a computed `getInvocationForId` to what we have now as well

Reply:
I think it'd be a significant improvement to move this all out of the onMounted hook. The computed looks good to me.

Reply:
We are now not computing/getting the workflow in the parent `WorkflowInvocationState` component at all
Instead, we use it in a child `WorkflowInvocationHeader` and the existing `WorkflowInvocationOverview` where we use `useWorkflowInstance` which expects an invocation.workflow_id` which we are ensuring exists in the parent.